File: app/src/main/java/com/keelim/cnubus/activities/MainAdapter.java
package com.keelim.cnubus.activities;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MainAdapter extends RecyclerView.Adapter<MainAdapter.PagerViewHolder> {
    private ArrayList<String> arrayList = null;

    @NonNull
    @Override
    public PagerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        View view = inflater.inflate(null, parent, false);
        MainAdapter.PagerViewHolder pagerViewHolder = new PagerViewHolder(view);
        return pagerViewHolder;
    }

    public MainAdapter(ArrayList<String> arrayList) {
        this.arrayList = arrayList;
    }

    @Override
    public void onBindViewHolder(@NonNull PagerViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return arrayList.size();
    }

    class PagerViewHolder extends RecyclerView.ViewHolder {

        public PagerViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/keelim/comssa/ui/main/MainAdapter.kt
package com.keelim.comssa.ui.main

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.keelim.comssa.data.db.entity.Search
import com.keelim.comssa.databinding.ItemSearchBinding

class MainAdapter(
    private val favoriteListener: (Int, Int) -> Unit,
): ListAdapter<Search, MainAdapter.ViewHolder>(diffUtil) {

    inner class ViewHolder(private val binding: ItemSearchBinding): RecyclerView.ViewHolder(binding.root){
        fun bind(item:Search) = with(binding){
            title.text = item.title
            description.text = item.description
            favoriteSwitch.isChecked = item.favorite == 1
            favoriteSwitch.setOnClickListener {
                favoriteListener.invoke(item.favorite, item.id)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(ItemSearchBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(currentList[position])
    }

    companion object{
        val diffUtil = object :DiffUtil.ItemCallback<Search>(){
            override fun areItemsTheSame(oldItem: Search, newItem: Search): Boolean {
                return newItem == oldItem
            }

            override fun areContentsTheSame(oldItem: Search, newItem: Search): Boolean {
                return newItem == oldItem
            }
        }
    }
}